components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -129,6 +129,20 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
             return authorizeRespDTO;
         }
 
+        List<String> allowedScopes = OAuthServerConfiguration.getInstance().getAllowedScopes();
+        List<String> requestedAllowedScopes = new ArrayList<>();
+        String[] requestedScope = authzReqMsgCtx.getAuthorizationReqDTO().getScopes();
+        List<String> requestedScopesList = new ArrayList<>();;
+        for (String scope : requestedScope) {
```

Comment:
Add null check for `requestedScope`. Also rename `requestedScope` to `requestedScopes`

Reply:
added in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1468/commits/a93b154c4c7b205e58ded516c4d53be85f268998